### PR TITLE
Comments to understand functions in the backend

### PR DESCRIPTION
**Pull request by cplkake**
Adds comments describing what the following functions are set up to accomplish above each respective function:

server/src/server.ts
- ifDefinedSet
- getUidForApiKey
- doApiKeyBasicAuth
- doApiKeyAuth
- doXidApiKeyAuth
- doHeaderAuth
- recordPermanentCookieZidJoin
- doXidConversationIdAuth
- _auth
  - getKey
  - doAuth
- enableAgid
- redirectIfHasZidButNoConversationId
- getXids
- handle_GET_xids
- doCookieAuth
- isOwner

server/src/session.ts
- getUserInfoForSessionToken

server/src/user.ts
- createDummyUser
- getXidRecordByXidOwnerId

### CODE REVIEW

File: server/src/server.ts [hunk 1]
```diff
@@ -234,6 +234,7 @@ function haltOnTimeout(req: { timedout: any }, res: any, next: () => void) {
   }
 }
 
+// checks if a property (name) exists in a source object and copies that property to a dest object
```
Comment (NewJerseyStyle) [inline server/src/server.ts]: To enable IDEs recognize the comment and shows the comment when developer hover on the function where it is been used, the following syntax would be better:

```ts
/**
 * Checks if a property exists in a source object and copies it to a destination object.
 *
 * @param name - The name of the property to check and copy.
 * @param source - The source object containing the property.
 * @param dest - The destination object to copy the property to.  If undefined, a new object will be created.
 * @returns The destination object with the property copied if it exists in the source object. Returns undefined if the source object is undefined.
 * @throws {Error} If the source object is not an object or the destination object is not an object or undefined.
 **/
```

More about TSDoc specification. [https://tsdoc.org/](https://tsdoc.org/)

File: server/src/server.ts [hunk 2]
```diff
@@ -2,7 +2,7 @@
 
 "use strict";
 
-import akismetLib from "akismet";
+import akismetLib from "akismet";   // spam protection for user-submitted text
```
Comment (NewJerseyStyle) [inline server/src/server.ts]: Adding a comment block at the beginning of your file to explain the purpose and key features of imported libraries is a good practice, especially when those libraries are not immediately obvious from the code itself or are less commonly used.

Here's how you might structure such a comment block at the beginning of the file, using the `akismet` example:
```ts
/**
 * @file This file contains the core logic for [brief description of the file's purpose].
 *
 * Dependencies:
 *  - `akismet`: This library is used for spam protection.  It provides functionality to check if user-submitted text is spam.  See [link to library documentation] for details on its API and usage.  Specifically, we're using it to [explain how you're using it, e.g., "check comments before posting"].
 *  - [Other library]: [brief explanation of its purpose and how it's used in this file].
 *
 *  Note: Ensure that the Akismet API key is correctly configured in the environment variables.  See [link to relevant documentation or configuration file].
 */

import { Client as AkismetClient } from "akismet";

// ... rest of your code ...
```

**Comment by NewJerseyStyle** [comment]
It is trivial to manually edit all the comment for each function, you can used AI to assist you drafting the comment block and then review, refine it before commit.

File: server/src/server.ts (hunk 1 above)
Comment (Zen-cronic) [reply server/src/server.ts]: Second this!

**Comment by patcon** [comment]
I'm going to close this for a sec, then re-open. that's just so I can try to sync the `edge` branch via the GitHub UI. Apparently creating PR's with commits from upstream confuses github and hides the "sync fork" option in the GitHub UI 🤷 

Screenshot 2024-10-29 at 12 07 59 AM

**Comment by patcon** [comment]
k that worked! I could click "sync fork" once I closed this. A bit wonky, but helpful to know :) Sorry for the noise in this PR

**Comment by NewJerseyStyle** [comment]
I guess it is because Venish was using `edge` from `compdemocracy` to make this PR? It doesn't break anything except confusing GitHub UI? 😂

**Comment by Zen-cronic** [comment]
After implementing the TSDoc, it's all good to merge! @cplkake

**Review by NewJerseyStyle — APPROVED** [review]
I have updated comment to make it follow TSDoc should be good to merge now

**Pull request merged**